src/model.c

```diff
@@ -236,6 +236,17 @@ RAI_Model *RAI_ModelCreate(RAI_Backend backend, RAI_Device device, int64_t devic
     RAI_SetError(err, RAI_EUNSUPPORTEDBACKEND, "Unsupported backend.\n");
     return NULL;
   }
+  if (device == RAI_DEVICE_CPU){
```

Comment:
This should be refactored: `devicestr` already provided to the command in `redisai.c` and parsed.
At this point, since we're dealing with strings as device/queue identifiers, the string should be passed directly to this function, which in turns delegates backends to accept or reject the string as a valid device.

This way the device can be made dynamic and backend-dependent (we can have MOVIDIUS:0 in ONNXRuntime).

I can do it in a follow-up PR right after this is merged.

Reply:
@lantiga done. please revise the changes on all backends for model_create and on torch for script_create